src/server/scripts/Commands/cs_misc.cpp

```diff
@@ -1384,9 +1384,30 @@ class misc_commandscript : public CommandScript
             target = PlayerIdentifier::FromTargetOrSelf(handler);
         }
 
-        if (!target || !target->IsConnected())
+        if (!target->IsConnected())
         {
-            return false;
+            if (handler->HasLowerSecurity(nullptr, target->GetGUID()))
+                return false;
+
+            QueryResult characterResult = CharacterDatabase.Query("SELECT `guid` FROM `characters` WHERE `name`='{}'", target->GetName());
+
+            if (!characterResult)
+                return false;
+
+            QueryResult result = CharacterDatabase.Query("SELECT `mapId`, `zoneId`, `posX`, `posY`, `posZ`, `posO` FROM `character_homebind` WHERE `guid`={}", (*characterResult)[0].Get<uint32>());
```

Comment:
There is a prepared statement for this query, `CHAR_SEL_CHARACTER_HOMEBIND`, and we should stick with prepared statements. But don't forget to change the connection type from `CONNECTION_ASYNC` to `CONNECTION_BOTH`.